editor/src/ui-tools/viewport/viewport-commands.ts

```diff
@@ -57,6 +58,37 @@ export class IvyFitToScreenCommand extends FitToScreenCommand {
     super(action);
   }
 
+  getNewViewport(bounds: Bounds, model: SModelRoot): Viewport | undefined {
+    if (!Dimension.isValid(model.canvasBounds)) {
+      return undefined;
+    }
+    const c = Bounds.center(bounds);
+    const delta = this.action.padding === undefined ? 0 : 2 * this.action.padding;
+    const toolBarHeight = this.toolBarHeight();
+    let zoom = Math.min(
+      model.canvasBounds.width / (bounds.width + delta),
+      (model.canvasBounds.height - toolBarHeight) / (bounds.height + delta)
+    );
+    if (this.action.maxZoom !== undefined) {
+      zoom = Math.min(zoom, this.action.maxZoom);
+    }
+    if (zoom === Infinity) {
+      zoom = 1;
+    }
+    return {
+      scroll: {
+        x: c.x - (0.5 * model.canvasBounds.width) / zoom,
+        y: c.y - (0.5 * (model.canvasBounds.height + toolBarHeight)) / zoom
+      },
+      zoom: zoom
+    };
+  }
+
+  private toolBarHeight(): number {
+    const toolBar = document.getElementById('sprotty_' + ToolBar.ID);
```

Comment:
I would hardcode it -> 48
If we change this we need to change a lot more anyway